### PR TITLE
support clear error with empty argument and array

### PR DESCRIPTION
**Pull request by bluebill1049**
**`clearError`: update**

- support clear all errors
- support clear sets of arrays

**`reset`: update**

- support pass reset defaultvalues

- [ ] update website docs

### CODE REVIEW

File: src/useForm.ts [hunk 1]
```diff
@@ -576,6 +584,11 @@ export default function useForm<
     validFieldsRef.current = new Set();
     submitCountRef.current = 0;
     isSchemaValidateTriggeredRef.current = false;
+    if (values) {
+      Object.entries(values).forEach(([key, value]) => {
+        setFieldValue(key as Name, value);
```
Comment (bluebill1049) [inline src/useForm.ts]: @barrymay @stramel please let me know if this is not ok.

**Comment by kabriel** [comment]
@bluebill1049 would love to see this merged

File: src/useForm.ts (hunk 1 above)
Comment (stramel) [reply src/useForm.ts]: This should be fine.

**Comment by bluebill1049** [comment]
@kabriel you want to give me a hand on testing? `3.21.0-beta.1` (I did a test on my end) I am working on `automation`, in the near future there will be less manually testing 👍

**Comment by kabriel** [comment]
doing some testing now. i have a few forms with automated testing, i'm using https://smashtest.io

**Comment by kabriel** [comment]
`reset` works perfectly, exactly what i needed. I didn't test `clearError` as i don't have a usecase for that (i clear all errors)

**Comment by bluebill1049** [comment]
Sweet! All good I will release it when I am in the office🥳

**Pull request merged**